### PR TITLE
week 17(revised) , 18(revised) and 19 algos

### PR DESCRIPTION
**Pull request by Robertito1**
week 17 algos

### CODE REVIEW

File: algos/level2/week17/robert.index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    constructor(){
+        this.values = []
+    }
+}
+
+class Queue {
+    constructor() {
+       const stackA = new Stack
+       const stackB = new Stack
+     this.values = stackA.values.concat(stackB.values)
```
Comment (gzandiemer) [inline algos/level2/week17/robert.index.js:12]: Hi Robert, stackA.values and stackB.values are undefined. That's why it won't work. Try to solve it without making an array from these 2 stacks. I know it will be very inefficient without an array, but we are just making an exercise :)  Try using already existing stack methods here within add(value), remove() and peek().

**Review by gzandiemer — CHANGES_REQUESTED** [review]

**Review by gzandiemer — COMMENTED** [review]
Hello Robert, this was a very challenging class to write. I myself had some issues and couldn't complete it alone. But you have done a good job. I tested your code with the tests in gzan.test.js file. 
Some of the methods are working but some not. Here I list the methods that cannot pass the test: getFirst, getLast, getAt, removeFirst, removeLast, removeAt, insertLast, insertAt. You can find the solutions to these methods in gzan.index.js file.

**Comment by Robertito1** [comment]
Thank you very Gzan, for the review I will find out why those ones are not passing the test using your own test suite and then adjust them accordingly

**Comment by Robertito1** [comment]
I have seen that the issue is because of the way I returned my data from those functions, when a node is added to the list I return it with the value that is added, for example getLast() = 66 // out is 66 where 66 is the data of the last node, but on your test file, it should return {data: 66, next: null}, the question did not say anything about how to handle outputs. I will fix this and also my tests accordingly.

**Comment by Robertito1** [comment]
All fixed, I looked more closely  at your test and found out that there were some unhandled edge cases in my code, thank you very much for this exercise.
